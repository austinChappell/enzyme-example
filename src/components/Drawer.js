// External Dependencies
import React, { Component } from 'react';
import styled from 'styled-components';

// Internal Dependencies
import { COLORS } from '../constants/styles';

// Local Variables
const StyledDiv = styled.div(props => ({
  alignItems: 'flex-start',
  backgroundColor: COLORS.primaryLight,
  boxShadow: '2px 2px 4px 1px #777',
  display: 'flex',
  height: '100vh',
  justifyContent: 'flex-end',
  left: props.isOpen ? 0 : -300,
  position: 'fixed',
  transition: '1000ms',
  width: 400,
}));

// Component Definition
class Drawer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false,
    };
  }

  toggleDrawer = () => {
    this.setState(state => ({ isOpen: !state.isOpen }));
  }

  render() {
    return (
      <StyledDiv isOpen={this.state.isOpen}>
        <button
          onClick={this.toggleDrawer}
          type="button"
        >
          {this.state.isOpen ? 'Close Drawer' : 'Open Drawer'}
        </button>
      </StyledDiv>
    );
  }
}

export default Drawer;
