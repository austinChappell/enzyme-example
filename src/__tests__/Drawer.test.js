import React from 'react';
import {
  mount,
  shallow,
} from 'enzyme';

import Drawer from '../components/Drawer';

describe('<Drawer />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <Drawer />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should mount in a closed state', () => {
    const wrapper = mount(
      <Drawer />,
    );
    expect(wrapper.state().isOpen).toBe(false);
  });

  it('should open when button is clicked', () => {
    const wrapper = mount(
      <Drawer />,
    );
    wrapper.find('button').simulate('click');
    expect(wrapper.state().isOpen).toBe(true);
  });

  it('should close if clicked from open state', () => {
    const wrapper = mount(
      <Drawer />,
    );
    wrapper.setState({ isOpen: true });
    wrapper.find('button').simulate('click');
    expect(wrapper.state().isOpen).toBe(false);
  });
});
