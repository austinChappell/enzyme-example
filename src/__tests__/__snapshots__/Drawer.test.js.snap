// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Drawer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <button
        onClick={[Function]}
        type="button"
      >
        Open Drawer
      </button>,
      "isOpen": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Open Drawer",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": "Open Drawer",
      "type": "button",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          [Function],
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <button
          onClick={[Function]}
          type="button"
        >
          Open Drawer
        </button>,
        "isOpen": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Open Drawer",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Open Drawer",
        "type": "button",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            [Function],
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
